### ISSUE TITLE
Add support for ITeardownSystem

### ISSUE THREAD

**Issue by grofit** [EcsRx/ecsrx.unity#27]
This would be the opposite of `ISetupSystem` so this can be used to indicate when an entity no longer applies to a group and can do custom logic on the entity in question.

**Comment by gregee123** [comment]
What condition would trigger that exactly? Is removing one component enough? would the system be notified what exact change to the entity triggered the tear-down?

**Comment by grofit** [comment]
The idea is that you provide it a group, and when something stops being part of that group (due to a component being removed) it would trigger the logic. This would allow you to have the ability to have complex view and binding setups in an `ISetupSystem` but then undo the setup in the `ITeardownSystem`. This would effectively allow you to have something similar to an `IManualSystem` but with it automatically handling new and removed entities automatically if you wanted it.

It would also improve use cases where systems have state pertaining to entities and you want to clean that up when the entity is no longer applicable.

```
public class CountingSystem : ISetupSystem, ITeardownSystem
{
   private int _localCount = 0;
   public IGroup TargetGroup { get { return new Group(typeof(CountedComponent)); } }

   public void Setup(IEntity entity) 
   { _localCount++;  }

   public void Teardown(IEntity entity)
   { _localCount--; }
}
```

Something like that although that is super simple it just allow you to handle similar use cases where you need to do *something* once the entity is no longer within a given group, again its not a mainstream concern, but it at least provides that functionality for those who do need to do this sort of thing.

**Comment by grofit** [comment]
This has been released in 0.8.0 feel free to re-open if there are any issues with this
